packages/web3-react/src/hooks/useConnect.tsx

```diff
@@ -96,27 +96,29 @@ export function useConnect(options: ConnectOptions) {
     }
   }, [context, wcDisconnect])
 
-  const disconnectAlephium = useCallback(() => {
+  const disconnectAlephium = () => {
     getDefaultAlephiumWallet()
       .then((alephium) => {
         if (!!alephium) {
           alephium.disconnect()
-          context.setAccount(undefined)
           context.setSignerProvider(undefined)
+          context.setAccount(undefined)
         }
       })
       .catch((error: any) => {
         console.error(error)
       })
-  }, [context])
+  }
 
   const connectAlephium = useCallback(async () => {
     const windowAlephium = await getDefaultAlephiumWallet()
 
     const enabledAccount = await windowAlephium
       ?.enable({
         ...options,
-        onDisconnected: disconnectAlephium
+        onDisconnected: () => {
```

Comment:
Question: What's the reason to set `onDisconnected` to no-op?

Reply:
Because we have already called `context.setSignerProvider(undefined)` and `context.setAccount(undefined)` at here: https://github.com/alephium/alephium-web3/blob/7cdf1fde136b81a459564c1c4b0285af86c4b96a/packages/web3-react/src/hooks/useConnect.tsx#L103

When the user clicks the disconnect button, the function call sequence is: `disconnectAlephium` -> `alephium.disconnect()` -> `onDisconnected`, which will cause `disconnectAlephium` to be executed twice. If I understand correctly, we only need to perform cleanup in either the `onDisconnect` callback or `disconnectAlephium`.

Reply:
Great point 👍, I just tested it, and the context will also be cleaned up in this case.